### PR TITLE
APM-391 Provide some guidance on how Apple Pay can be integrated directly

### PR DESCRIPTION
**Pull request by willemstuursma**
@ndijkstra @shirshir @MollieRick @albert-carreras Would be great to get some advice / tips from you. 

@albert-carreras do we have any stock pictures of how it'd look on you own checkout?

### CODE REVIEW

File: source/wallets/applepay.rst [hunk 1]
```diff
@@ -204,8 +204,25 @@ For more details on detecting the support of Apple Pay on a device, see `Apple
...
+
+#. Check if Apple Pay is `available on the device <https://developer.apple.com/documentation/apple_pay_on_the_web/apple_pay_js_api/checking_for_apple_pay_availability>`_
+   using the `canMakePayments <https://developer.apple.com/documentation/apple_pay_on_the_web/applepaysession/1778027-canmakepayments>`_ method.
+
+#. Create an `Apple Pay Session <https://developer.apple.com/documentation/apple_pay_on_the_web/apple_pay_js_api/creating_an_apple_pay_session>`_.
+
+#. `Provide merchant validation <https://developer.apple.com/documentation/apple_pay_on_the_web/apple_pay_js_api/providing_merchant_validation>`_.
+   Instead of using Apple's API, you can use a :doc:`dedicated endpoint in the  Mollie API </reference/v2/miscellaneous/request-apple-pay-payment-session>`.
+
+#. One the shopper has authorized the payment, you will receive the `Apple Pay Payment object <https://developer.apple.com/documentation/apple_pay_on_the_web/applepaypayment>`_.
```
Comment (albert-carreras) [inline source/wallets/applepay.rst]: Once

File: source/reference/v2/miscellaneous/request-apple-pay-payment-session.rst [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+``https://[domain]/.well-known/apple-developer-merchantid-domain-association``. Without this file, it will not be
+possible to use Apple Pay on your domain.
+
+The guidelines for working with a payment session are:
+
+- Request a new payment session object for each transaction. You can only use a merchant session object a single time.
+
+- The payment session object expires five minutes after it is created.
+
+- Never request the payment session from the browser. The request must be sent from your server.
```
Comment (MollieRick) [inline source/reference/v2/miscellaneous/request-apple-pay-payment-session.rst]: Maybe creating a warningbox (`... warning:`) for it?

File: source/wallets/applepay.rst [hunk 3]
```diff
@@ -204,8 +204,25 @@ For more details on detecting the support of Apple Pay on a device, see `Apple
...
+   You must then add the `token <https://developer.apple.com/documentation/apple_pay_on_the_web/applepaypaymenttoken>`_
+   when invoking the :doc:`Create Payment API </reference/v1/payments-api/create-payment>`.
```
Comment (ndijkstra) [inline source/wallets/applepay.rst]: v1 Payments API?

File: source/reference/v2/miscellaneous/request-apple-pay-payment-session.rst [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+   * - ``validationUrl``
+
+       .. type:: string
+          :required: true
+
+     - The ``validationUrl`` you got from the `ApplePayValidateMerchant event <https://developer.apple.com/documentation/apple_pay_on_the_web/applepayvalidatemerchantevent>`_.
+
+       A `list of all valid domain names <https://developer.apple.com/documentation/apple_pay_on_the_web/setting_up_your_server#3172427>`_
+       for merchant validation is available. You should white list these in your application and reject any
```
Comment (ndijkstra) [inline source/reference/v2/miscellaneous/request-apple-pay-payment-session.rst]: Maybe make the white list part more clear?

File: source/reference/v2/miscellaneous/request-apple-pay-payment-session.rst [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+       ``validationUrl`` not in the list.
+
+   * - ``domain``
```
Comment (ndijkstra) [inline source/reference/v2/miscellaneous/request-apple-pay-payment-session.rst]: Call this `domainName` like in the response?

File: source/wallets/applepay.rst (hunk 3 above)
Comment (willemstuursma) [reply source/wallets/applepay.rst]: Thanks

File: source/reference/v2/miscellaneous/request-apple-pay-payment-session.rst (hunk 5 above)
Comment (willemstuursma) [reply source/reference/v2/miscellaneous/request-apple-pay-payment-session.rst]: I looked at it, but Apple uses the term _domain_ (like in the filename and all the communication I got from Apple). So I'd like to stick with that.

**Comment by mattijsbliek** [comment]
> `do we have any stock pictures of how it'd look on you own checkout?`
> I think @mattijsbliek designed a few of those screens and could provide them

Those contain copyrighted images, would have to tweak a bit to allow use in our docs. Will look at it as well as this PR somewhere next week.

**Comment by willemstuursma** [comment]
@mattijsbliek I copied a picture from Apple's promo site for now

File: source/theme/sidebar-reference-v2.html [hunk 6]
```diff
@@ -492,3 +492,15 @@
...
 </div>
+
+<div class="sidebar__group {{ sidebar_collapse_group('reference/v2/methods-api/') }}">
+	<p><a href="javascript:void(0)">Miscellaneous</a></p>
+
+	<ul>
+		{{ sidebar_item(
+			'reference/v2/miscellaneous/request-apple-pay-payment-session',
+			'Request Apple Pay Payment Session',
+			'POST /v2/request-apple-pay-payment-session'
```
Comment (willemstuursma) [inline source/theme/sidebar-reference-v2.html]: @shirshir what do you think of this endpoint?

Comment (shirshir) [reply source/theme/sidebar-reference-v2.html]: What about `POST /v2/wallets/apple-pay/create-session`?

**Pull request merged**